pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1567,7 +1567,8 @@ public CompletableFuture<ManagedLedgerConfig> getManagedLedgerConfig(TopicName t
                     RetentionPolicies retentionPolicies = null;
                     OffloadPoliciesImpl topicLevelOffloadPolicies = null;
 
-                    if (pulsar.getConfig().isTopicLevelPoliciesEnabled()) {
+                    if (pulsar.getConfig().isTopicLevelPoliciesEnabled()
+                            && !NamespaceService.isSystemServiceNamespace(namespace.toString())) {
```

Comment:
Maybe we can also move this check `!NamespaceService.isSystemServiceNamespace(namespace.toString())` to  `pulsar.getTopicPoliciesService().getTopicPolicies(topicName)`?

Because I see you already check it at `SystemTopicBasedTopicPoliciesService#sendTopicPolicyEvent`.

Reply:
Ah, we should avoid creating the topic first. This is the root cause.